pkg/controller/controller.go

```diff
@@ -149,6 +150,7 @@ func (c *Controller) Run() error {
 				// TODO: set version in spec in apiserver
 				s.Version = defaultVersion
 			}
+			clus.Spec.Version = strings.Trim(clus.Spec.Version, "v")
```

Comment:
spec should be "readonly". it should be the user input. i think your previous approach is better. just need some tests.